core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1432,6 +1432,7 @@ class PreferredReplicaElectionListener(protected val controller: KafkaController
   def doHandleDataDeleted(dataPath: String) {}
 }
 
+//TODO: Very dangerous, if put into map as key and later any var will change, magic will happen!!!
```

Comment:
nit: we can change this comment to

```
/**
 * Note that this class cannot be used as map keys since its fields are vars"
 */
```

Another code-style meta comment is to use `//` for in-function comments and `/**` for outside function comments, and we use a space as indent.